### PR TITLE
pynmeagps: init at 1.0.35

### PR DESCRIPTION
**Pull request by dylan-gonzalez**
## Description of changes
Add the pynmeagps python package

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pynmeagps/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    pytest-cov
+  ];
+
+  pythonImportsCheck = [
+    "pynmeagps"
+  ];
+
+  meta = {
+    homepage = "https://github.com/Knio/pynmea2";
+    description = "Python 3 parser aimed primarily at the subset of the NMEA 0183 v4 protcol relevant to GNSS/GPS receivers.";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/pynmeagps/default.nix:34]:
```suggestion
    description = "Python 3 parser aimed primarily at the subset of the NMEA 0183 v4 protcol relevant to GNSS/GPS receivers";
```

File: pkgs/development/python-modules/pynmeagps/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+, fetchFromGitHub
+, pytestCheckHook
+, setuptools
+, pytest-cov
+}:
+
+buildPythonPackage rec {
+  pname = "pynmeagps";
+  version = "1.0.35";
+  format = "pyproject";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/pynmeagps/default.nix:12]:
```suggestion
  pyproject = true;
```

File: pkgs/development/python-modules/pynmeagps/default.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  src = fetchFromGitHub {
+    owner = "semuconsulting";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/pynmeagps/default.nix:16]:
```suggestion
    repo = "pynmeagps";
```

Comment (dylan-gonzalez) [reply pkgs/development/python-modules/pynmeagps/default.nix:16]: is that the standard?

**Comment by arduano** [comment]
Result of `nixpkgs-review pr 302260` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pynmeagps</li>
    <li>python311Packages.pynmeagps.dist</li>
    <li>python312Packages.pynmeagps</li>
    <li>python312Packages.pynmeagps.dist</li>
  </ul>
</details>

**Comment by arduano** [comment]
Make sure you squash the commits btw

**Comment by dylan-gonzalez** [comment]
> Make sure you squash the commits btw
sorry, done now

**Comment by dylan-gonzalez** [comment]
Not sure why the ofborg-eval-lib-tests is failing when it didn't before?

File: pkgs/development/python-modules/pynmeagps/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    pytestCheckHook
+    pytest-cov
+  ];
+
+  pythonImportsCheck = [
+    "pynmeagps"
+  ];
+
+  meta = {
+    homepage = "https://github.com/Knio/pynmea2";
```
Comment (sikmir) [inline pkgs/development/python-modules/pynmeagps/default.nix:33]:
```suggestion
    homepage = "https://github.com/semuconsulting/pynmeagps";
```

File: pkgs/development/python-modules/pynmeagps/default.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    description = "Python 3 parser aimed primarily at the subset of the NMEA 0183 v4 protcol relevant to GNSS/GPS receivers";
```
Comment (sikmir) [inline pkgs/development/python-modules/pynmeagps/default.nix:34]:
```suggestion
    description = "NMEA protocol parser and generator";
```

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 302260` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.pynmeagps</li>
    <li>python311Packages.pynmeagps.dist</li>
    <li>python312Packages.pynmeagps</li>
    <li>python312Packages.pynmeagps.dist</li>
  </ul>
</details>

**Review by sikmir — APPROVED** [review]

File: pkgs/development/python-modules/pynmeagps/default.nix (hunk 3 above)
Comment (sikmir) [reply pkgs/development/python-modules/pynmeagps/default.nix]: No, `repo = pname` is an antipattern.

**Pull request merged**